### PR TITLE
feat(wallets): add support for defly wallet

### PR DESCRIPTION
**Pull request by envin3**

### CODE REVIEW

File: package.json
Comment (oliviera9) [inline package.json:1]: Dapp version has not been updated

File: src/lib/algoWalletModal/connectors/defly-wallet.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  close() {
+    this.deflyWallet.disconnect()
+    return
+  }
+
+  async signTxn(_txns) {
+    const a = await this.deflyWallet.signTransaction([_txns.map((_tx) => ({ txn: _tx }))])
+    return a.map((_a) => Buffer.from(_a).toString('base64'))
```
Comment (oliviera9) [inline src/lib/algoWalletModal/connectors/defly-wallet.js:19]: Did you have a try?

File: src/lib/algoWalletModal/connectors/defly-wallet.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  }
+}
+
+const connectToDeflyWallet = async () => {
+  const deflyWallet = new DeflyWalletConnect()
+  let accounts = []
+  try {
+    accounts = await deflyWallet.connect()
+  } catch (err) {
+    accounts = await deflyWallet.reconnectSession()
```
Comment (oliviera9) [inline src/lib/algoWalletModal/connectors/defly-wallet.js:33]: I would check the error message? How can you be sure the reconnectSession will work?

File: src/lib/algoWalletModal/index.jsx [hunk 3]
```diff
@@ -78,6 +79,17 @@ class AlgoWalletModal extends EventEmitter {
       },
     })
 
+    this.userOptions.push({
```
Comment (oliviera9) [inline src/lib/algoWalletModal/index.jsx:82]: 👍

File: src/lib/algoWalletModal/connectors/defly-wallet.js (hunk 1 above)
Comment (envin3) [reply src/lib/algoWalletModal/connectors/defly-wallet.js:19]: I tried to signed a tx

Comment (oliviera9) [reply src/lib/algoWalletModal/connectors/defly-wallet.js:19]: I was wondering if the result of signTransaction really needs to be base64 encoded, and/or if it an array of strings, or an array of object from where you need to retrieve strings.

File: src/lib/algoWalletModal/connectors/defly-wallet.js (hunk 2 above)
Comment (envin3) [reply src/lib/algoWalletModal/connectors/defly-wallet.js]: updated

**Pull request merged**